app/src/main/java/org/mozilla/fenix/RunBlockingCounter.kt

```diff
@@ -0,0 +1,40 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// This class implements the alternative ways to invoke runBlocking with some
+// monitoring by wrapping the raw methods. This lint check tells us not to use the raw
+// methods so we suppress the check.
+@file:Suppress("MozillaRunblockingCheck")
+
+package org.mozilla.fenix
+
+import androidx.annotation.VisibleForTesting
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.runBlocking
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.EmptyCoroutineContext
+
+/**
+ * Counts the number of runBlocking calls made
+ */
+@VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+class RunBlockingCounter {
+    companion object {
+        var runBlockingCount = 0L
+    }
+}
+
+/**
+ * Wrapper around `runBlocking`
+ */
+fun <T> runBlockingIncrement(
```

Comment:
I don't love this name. While it's telling us what it does, it's also kind of vague – increment what?

Naming this seems really hard. The intentions I can think of are:
- tell them what it does (*increment, *perf, *monitor, ...)
- tell them it's preferred (*safe, *preferred, ...)

Maybe `runBlockingMonitored`? It sounds intimidating so maybe people will think twice before using it. 🤣 Or maybe they'll think we're the perf 👮‍♀️ :(

Reply:
I guess that can work. I was thinking  that 1) if the kdoc explains it further in details, then a looser name could be fine? Or 2) we could go longer, but more precise such as `runBlockingPerformanceTracker`? It seems a bit less intimidating and its purpose is conveyed a lot more from the name.

Reply:
I think loose names are only okay if the function is frequently used, e.g. `map` or `add`. Given this is a function people aren't supposed to call a lot anyway, I think it's fine to have a longer name. However, I wanted to avoid making it too tied to performance so people don't think, "perf police!"